pygmsh/opencascade/geometry.py

```diff
@@ -129,33 +146,33 @@ def _boolean_operation(
                 operation,
                 legal_dim_types[dim_type],
                 ','.join(e.id for e in input_entities),
-                'Delete;' if delete else '',
+                'Delete;' if delete1 else '',
                 legal_dim_types[dim_type],
                 ','.join(e.id for e in tool_entities),
-                'Delete;' if delete else ''
+                'Delete;' if delete2 else ''
                 ))
 
         return dim_type(id0=name, is_list=True)
 
-    def boolean_intersection(self, entities, delete=True):
+    def boolean_intersection(self, entities, delete1=True, delete2=True):
```

Comment:
We'll have to think about the interface here. One supplies a list of entities to `boolean_intersection`, and it's not entirely clear to the user what `delete1` and `delete2` mean in the context. Perhaps `delete_first` and `delete_other` would be more appropriate.

Reply:
Good point thanks. Fixed.